src/ViewWrapper.js

```diff
@@ -0,0 +1,192 @@
+    const [brushStartX, setBrushStartX] = useState(null);
+    const [brushEndX, setBrushEndX] = useState(null);
+    
+    const [requestedIntervalParams, setRequestedIntervalParams] = useState(null);
+
+    const brushBarHeight = 24;
+
+    // Set up the d3-drag handler functions (started, ended, dragged).
+    const started = useCallback(() => {
+        const div = divRef.current;
+        const [mouseX, mouseY] = d3.mouse(div);
+
+        dragX.current = mouseX;
+        newViewportId.current = uuidv4();
+        
+        setBrushStartX(mouseX);
+        setMouseHoverX(null);
+    }, [dragX]);
+
+    const dragged = useCallback(() => {
+        const [mouseX, mouseY] = d3.mouse(divRef.current);
+
+        const {start, end} = getRange(dragX.current, mouseX, 0, viewBoundingBox.width);
+        
+        setBrushStartX(start);
+        setBrushEndX(end);
+    }, [dragX]);
+
+    const ended = useCallback(() => {
+        const [mouseX, mouseY] = d3.mouse(divRef.current);
+
+        const { start, end } = getRange(dragX.current, mouseX, 0, viewBoundingBox.width);
+
+        let startProp = start / viewBoundingBox.width;
+        let endProp = end / viewBoundingBox.width;
+        
+        if(Math.abs(end - start) > 4) {
+            onSelectGenomicInterval(startProp, endProp, newViewportId.current);
+        }
+
+        dragX.current = null;
+        setBrushEndX(0);
+        setBrushStartX(null);
+    }, [dragX, brushStartX, brushEndX, viewBoundingBox]);
+
+    useEffect(() => {
+        const div = divRef.current;
+
+        const drag = d3.drag()
+            .on("start", started)
+            .on("drag", dragged)
+            .on("end", ended);
+        
+        d3.select(div).call(drag);
+
+        return () => d3.select(div).on(".drag", null);
+    });
+
+    useEffect(() => {
+        const div = divRef.current;
+
+        d3.select(div).on("mousemove", () => {
+            const [mouseX, mouseY] = d3.mouse(div);
+            setMouseHoverX(mouseX);
+        });
+        
+        d3.select(div).on("mouseleave", () => {
+            setMouseHoverX(null);
+        });
+
+        return () => {
+            d3.select(div).on("mousemove", null);
+            d3.select(div).on("mouseleave", null);
+        };
+    });
+
+    // All hooks must be above this return statement, since they need to be executed in the same order.
+    if(!viewBoundingBox || !multivecTrack || !multivecTrack.tilesetInfo) {
+        // The view info has not yet loaded.
+        return null;
+    }
+
+    const { top, left, width, height } = viewBoundingBox;
+    const brushBarTop = height + 4;
+
+    return (
+        <div className="cistrome-hgw-view-wrapper"
+            style={{
+                position: "absolute",
+                top: `${top}px`,
+                left: `${left}px`,
+                width: `${width}px`, 
+                height: `${height}px`,
+                pointerEvents: "none"
+        }}>
```

Comment:
```suggestion
        <div 
            className="cistrome-hgw-view-wrapper"
        >
            <div 
                style={{
                    position: "absolute",
                    top: `${top}px`,
                    left: `${left}px`,
                    width: `${width}px`, 
                    height: `${height}px`,
                    pointerEvents: "none"
                }}
            >
```

I think there should be another level to the div structure here, since the table should be contained in the view-wrapper div but should be outside of this top, left, width, height rectangle.

Reply:
Great point. Definitely should be on the outside.